api/resourcecreator.go

```diff
@@ -607,6 +620,52 @@ func createService(deploymentRequest NaisDeploymentRequest, k8sClient kubernetes
 	return createServiceResource(serviceDef, deploymentRequest.Namespace, k8sClient)
 }
 
+func createRedisDef(deploymentRequest NaisDeploymentRequest) *redisapi.RedisFailover {
+	replicas := int32(3)
+	resources := redisapi.RedisFailoverResources{
+		Limits:   redisapi.CPUAndMem{Memory: "100Mi"},
+		Requests: redisapi.CPUAndMem{CPU: "100m"},
+	}
+	if deploymentRequest.FasitEnvironment != ENVIRONMENT_P {
+		replicas = int32(1)
+		resources = redisapi.RedisFailoverResources{
+			Limits:   redisapi.CPUAndMem{Memory: "50Mi"},
+			Requests: redisapi.CPUAndMem{CPU: "50m"},
+		}
+	}
+
+	spec := redisapi.RedisFailoverSpec{
+		HardAntiAffinity: false,
+		Sentinel: redisapi.SentinelSettings{
+			Replicas:  replicas,
+			Resources: resources,
+		},
+		Redis: redisapi.RedisSettings{
+			Replicas:  replicas,
+			Resources: resources,
+			Exporter:  true,
+		},
+	}
+	meta := k8smeta.ObjectMeta{Name: deploymentRequest.Application, Namespace: deploymentRequest.Namespace}
+	return &redisapi.RedisFailover{Spec: spec, ObjectMeta: meta}
+}
+
+func createRedisFailover(deploymentRequest NaisDeploymentRequest) (*redisapi.RedisFailover, error) {
+	failover := createRedisDef(deploymentRequest)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, fmt.Errorf("can't create InClusterConfig: %s", err)
+	}
+
+	client, err := redisclient.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("can't create new Redis client for InClusterConfig: %s", err)
+	}
+
+	return redisclient.RedisFailoversGetter(client).RedisFailovers(deploymentRequest.Namespace).Create(failover)
```

Comment:
Should there be a check for an existing redis instance? What will happen if an  existing redis instance is present?

Reply:
Yes it should. Not sure what will happen if it already exists. I'll try it out.

Reply:
This is the error message you get. So yeah, should check first.

>>> failed while creating or updating k8s-resources: failed while creating Redis failover: redisfailovers.storage.spotahome.com "nais-testapp" already exists (500)